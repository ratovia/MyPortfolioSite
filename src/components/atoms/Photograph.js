import React from "react"
import styled from "styled-components"

const RadiusImage = styled.img`
  border-radius: 25px;
`

export const Photograph = ({ imageUrl, radius }) => {
  return (
    <>
      {(() => {
        if (radius) {
          return (
            <RadiusImage
              src={imageUrl}
              alt="アイキャッチ"
              width="100%"
              height="100%"
            ></RadiusImage>
          )
        } else {
          return (
            <img src={imageUrl} alt="アイコン" width="100%" height="100%"></img>
          )
        }
      })()}
    </>
  )
}
