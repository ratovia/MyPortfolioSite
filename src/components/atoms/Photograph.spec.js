import React from "react"
import renderer from "react-test-renderer"
import { Photograph } from "./Photograph"
import reactIcon from "../../stories/assets/react_icon.jpg"

describe("Photograph", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Photograph imageUrl={reactIcon} radius />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders correctly", () => {
    const tree = renderer.create(<Photograph imageUrl={reactIcon} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
