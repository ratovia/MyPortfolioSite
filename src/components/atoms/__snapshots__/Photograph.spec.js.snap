// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photograph renders correctly 1`] = `
<img
  alt="アイキャッチ"
  className="sc-bdvvaa fWZaxb"
  height="100%"
  src="test-file-stub"
  width="100%"
/>
`;

exports[`Photograph renders correctly 2`] = `
<img
  alt="アイコン"
  height="100%"
  src="test-file-stub"
  width="100%"
/>
`;
